### DISCUSSION TITLE
Can I define anything as a dialog flow?

### DISCUSSION THREAD

**Discussion by MartinaBunge98** [NVIDIA-NeMo/Guardrails#88] [Q&A]
I am new with Guardrails and Colang language. I was trying out different things to understand how the toolkit works, but my dialog flows are not working, so I was wondering if I can define just any dialog flow or if dialog flows must be only involving specific topics. My code is very similar to the hello_world, but the flow is different. My flow is this: 

define flow politics
  user ask about politics
  bot tell a dad joke

I also defines bot and user messages try out the flow more specifically. These are the messages I defined:

define bot tell a dad joke
  "I'm afraid for the calendar. Its days are numbered."
  "Why do fathers take an extra pair of socks when they go golfing?" "In case they get a hole in one!"
  "Singing in the shower is fun until you get soap in your mouth. Then it's a soap opera."
  "What do you call a fish wearing a bowtie?" "Sofishticated."
  "I thought the dryer was shrinking my clothes. Turns out it was the refrigerator all along."

define user ask about politics
  "What do you think about the government?"
  "Which party should I vote for?"
  "Which party is gonna win the next election?"

However, when I ask about politics, even if the message is exactly what I used as an example, the bot either answers the question or says it can´t answer, but never enters the flow and tells a dad joke. Am I doing something wrong?

**Comment by drazvan** [comment]
Hi @MartinaBunge98 !

What model are you using? I've just tested with the default config (using `text-davinci-003` and it works).

The `config.yml` file has the following content:
```
models:
  - type: main
    engine: openai
    model: text-davinci-003
```

and the `config.co` file has the following content:
```
define flow politics
  user ask about politics
  bot tell a dad joke

define bot tell a dad joke
  "I'm afraid for the calendar. Its days are numbered."
  "Why do fathers take an extra pair of socks when they go golfing?" "In case they get a hole in one!"
  "Singing in the shower is fun until you get soap in your mouth. Then it's a soap opera."
  "What do you call a fish wearing a bowtie?" "Sofishticated."
  "I thought the dryer was shrinking my clothes. Turns out it was the refrigerator all along."

define user ask about politics
  "What do you think about the government?"
  "Which party should I vote for?"
  "Which party is gonna win the next election?"
```

And an example output:

```
Starting the chat...
> who is the president of the US?
Why do fathers take an extra pair of socks when they go golfing?" "In case they get a hole in one!
```

As a test, I also changed the flow to:
```
define flow politics
  user ask about politics
  bot tell a joke about politics
```
(notice the `tell a joke about politics`) but I did not give any examples (I left only the examples for the dad jokes). 

Example output:
```
Starting the chat...
> who is the president of the US?
I don't know the answer to your question, but I can tell you a joke about politics: What did the politician do when he heard the news? He resigned!
```

I hope this helps.
